runtime/src/main/java/com/dylibso/chicory/runtime/Store.java

```diff
@@ -0,0 +1,166 @@
+                .addTable(hostImports.tables());
+    }
+
+    /**
+     * Convert the contents of a store to a {@link HostImports} instance.
+     */
+    public HostImports toHostImports() {
+        return new HostImports(
+                functions.values().toArray(new HostFunction[0]),
+                globals.values().toArray(new HostGlobal[0]),
+                memories.values().toArray(new HostMemory[0]),
+                tables.values().toArray(new HostTable[0]));
+    }
+
+    /**
+     * Register an instance in the store with the given name.
+     * All the exported functions, globals, memories, and tables are added to the store
+     * with the given name.
+     *
+     * For instance, if a module named "myModule" exports a function
+     * named "myFunction", the function will be added to the store with the name "myFunction.myModule".
+     *
+     */
+    public Store register(String name, Instance instance) {
+        ExportSection exportSection = instance.module().exportSection();
+        for (int i = 0; i < exportSection.exportCount(); i++) {
+            Export export = exportSection.getExport(i);
+            String exportName = export.name();
+            switch (export.exportType()) {
+                case FUNCTION:
+                    ExportFunction f = instance.export(exportName);
+                    FunctionType ftype = instance.exportType(exportName);
+                    this.addFunction(
+                            new HostFunction(
+                                    (inst, args) -> f.apply(args),
+                                    name,
+                                    exportName,
+                                    ftype.params(),
+                                    ftype.returns()));
+                    break;
+
+                case TABLE:
+                    this.addTable(new HostTable(name, exportName, instance.table(export.index())));
+                    break;
+
+                case MEMORY:
+                    this.addMemory(new HostMemory(name, exportName, instance.memory()));
+                    break;
+
+                case GLOBAL:
+                    GlobalInstance g = instance.global(export.index());
+                    this.addGlobal(new HostGlobal(name, exportName, g));
+                    break;
+            }
+        }
+        return this;
+    }
+
+    /**
+     * A shorthand for instantiating a module and registering it in the store.
+     */
+    public Instance instantiate(String name, Module m) {
+        HostImports hostImports = toHostImports();
+        Instance instance = Instance.builder(m).withHostImports(hostImports).build();
+        register(name, instance);
+        return instance;
+    }
+
+    /**
+     * QualifiedName is internally used to use pairs (moduleName, name) as keys in the store.
+     */
+    static class QualifiedName {
+        private final String moduleName;
+        private final String name;
```

Comment:
nitpick: I would call it `fieldName` or something, there are already enough "name"s around here :sweat_smile:

Reply:
I'll use `fieldName` for consistency, but I feel we should revisit naming names ;)

Reply:
be my guest :pray: